### PR TITLE
Panels: adding box-shadow for panels with stretched-link

### PR DESCRIPTION
**Pull request by Garneauma**

### CODE REVIEW

**Comment by duboisp** [comment]
Can you provide a working example or describe where we can test it?

**Comment by Garneauma** [comment]
> Can you provide a working example or describe where we can test it?

Demo added.

**Review by GormFrank — COMMENTED** [review]
Suggestion : En plus des `panel`, nous pourrions ajouter le drop shadow aux `well`, aux `alert`, et même aux éléments qui ont la combinaison de classes `.brdr-bttm,.brdr-lft,.brdr-rght,.brdr-tp`. 

Il faudra aussi ajouter des exemples pratiques pour chacun.

**Comment by GormFrank** [comment]
This will fix the accessibility concern (apart from :visited) related to stretched link usage within any kinds of box layouts (cards, panels, alerts, etc.) @Garneauma

**Comment by Garneauma** [comment]
Box shadow now applied to `.well` and `.panel` elements containing a `.stretched-link` AND are not inside another `.well` or `.panel`. Discussed with @GormFrank and we decided to not include support for `.brdr-bttm,.brdr-lft,.brdr-rght,.brdr-tp`.

**Comment by duboisp** [comment]
Pre-approve upon review and local testing.

File: src/other/utility/utility-en.hbs [hunk 1]
```diff
@@ -502,15 +504,47 @@
 		<h3 class="h5"><a href="#" class="stretched-link">[Hyperlink text]</a></h3>
 		<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
 	</div>
+	<div class="col-md-4">
+		<div class="panel panel-default position-relative">
+			<div class="panel-body">
+				<img src="https://via.placeholder.com/360x203/000000/FFFFFF.png" alt="" class="img-responsive thumbnail mrgn-bttm-sm">
+				<h3 class="h5"><a href="#" class="stretched-link">[Hyperlink text]</a></h3>
```
Comment (duboisp) [inline src/other/utility/utility-en.hbs:511]: Panel should not be used alone to simply created a bordered box. That is a miss use of that pattern.

```suggestion
		<section class="panel panel-default position-relative">
		<header class="panel-heading">
   <h3 class="panel-title"><a href="#" class="stretched-link">[Hyperlink text]</a></h3>
   </header>
			<div class="panel-body">
				<img src="https://via.placeholder.com/360x203/000000/FFFFFF.png" alt="" class="img-responsive thumbnail mrgn-bttm-sm">
```

File: src/other/utility/utility-en.hbs [hunk 2]
```diff
@@ -502,15 +504,47 @@
...
+			</div>
+		</div>
+	</div>
+	<div class="col-md-4">
+		<div class="well position-relative">
+			<img src="https://via.placeholder.com/360x203/000000/FFFFFF.png" alt="" class="img-responsive thumbnail mrgn-bttm-sm">
+			<h3 class="h5"><a href="#" class="stretched-link">[Hyperlink text]</a></h3>
+			<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
+		</div>
+	</div>
```
Comment (duboisp) [inline src/other/utility/utility-en.hbs:522]: A section should be used to semantically connect the image along with its related heading.

```suggestion
	<div class="col-md-4">
		<section class="well position-relative">
			<img src="https://via.placeholder.com/360x203/000000/FFFFFF.png" alt="" class="img-responsive thumbnail mrgn-bttm-sm">
			<h3 class="h5"><a href="#" class="stretched-link">[Hyperlink text]</a></h3>
			<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
		</section>
	</div>
```

File: src/base/_component.scss [hunk 3]
```diff
@@ -37,6 +37,17 @@ button {
...
 
+/*
+Panels
+ */
+.panel, .well {
+	&:not(:has(.panel, .well)):has(.stretched-link) {
+		&:hover {
+			box-shadow: 1px 5px 7px rgba(0,0,0,.15);
+		}
+	}
```
Comment (duboisp) [inline src/base/_component.scss:48]: This don't work in FF only in Chrome. Do we have a little FF equivalent? Maybe a lighter effect?

**Review by duboisp — CHANGES_REQUESTED** [review]
This provide effect on hover, should we add a similar effect on focus and/or active event?

File: src/base/_component.scss (hunk 3 above)
Comment (Garneauma) [reply src/base/_component.scss:48]: There is no way to achieve this styling without ":has()" selector. If the styling HAS to work on Firefox, I'll have to put this PR on hold until Firefox is up to date with every other browser.

**Comment by Garneauma** [comment]
> This provide effect on hover, should we add a similar effect on focus and/or active event?

Fixed all issues and added styling on focus.

**Review by duboisp — APPROVED** [review]
Reviewed and tested locally, work as expected.

Considering this is only a style and not a functionality, then it is fine to have it for Chrome and not FF. There is no lost of functionality. We do know that FF are going to support the CSS `:has` in the coming month.

**Pull request merged**